reVX/plexos/rev_reeds_plexos.py

```diff
@@ -1,1460 +1,1134 @@
+            res_gids, res_id_counts.
+        reeds_build : pd.DataFrame
+            REEDS buildout with rows for built capacity at each reV SC point.
+
+        Returns
+        -------
+        table : pd.DataFrame
+            rev_sc and reeds_build inner joined on supply curve gid.
+        """
+
+        if 'gid' in rev_sc and 'gid' in reeds_build:
+            rev_join_on = 'gid'
+            reeds_join_on = 'gid'
+        else:
+            raise KeyError('GID must be in reV SC and REEDS Buildout tables!')
+
+        rev_sc, reeds_build = self._check_rev_reeds_coordinates(
+            rev_sc, reeds_build, rev_join_on, reeds_join_on)
+
+        check_isin = np.isin(reeds_build[reeds_join_on].values,
+                             rev_sc[rev_join_on].values)
+        if not all(check_isin):
+            wmsg = ('There are REEDS buildout GIDs that are not in the reV '
+                    'supply curve table: {} out of {} total REEDS buildout '
+                    'sites.'.format(np.sum(~check_isin), len(reeds_build)))
+            warn(wmsg)
+            logger.warning(wmsg)
+
+        table = pd.merge(rev_sc, reeds_build, how='inner', left_on=rev_join_on,
+                         right_on=reeds_join_on)
+
+        return table
+
+    @staticmethod
+    def _check_rev_reeds_coordinates(rev_sc, reeds_build, rev_join_on,
+                                     reeds_join_on, threshold=0.5):
+        """Check that the coordinates are the same in rev and reeds buildouts.
+
+        Parameters
+        ----------
+        rev_sc : pd.DataFrame
+            reV supply curve results table including SC gid, lat/lon,
+            res_gids, res_id_counts.
+        reeds_build : pd.DataFrame
+            REEDS buildout with rows for built capacity at each reV SC point.
+        rev_join_on : str
+            Column name to join rev table.
+        reeds_join_on : str
+            Column name to join reeds table.
+        threshold : float
+            Maximum difference in coord matching.
+
+        Returns
+        -------
+        rev_sc : pd.DataFrame
+            Same as input.
+        reeds_build : pd.DataFrame
+            Same as input but without lat/lon columns if matched.
+        """
+
+        rev_coord_labels = get_coord_labels(rev_sc)
+        reeds_coord_labels = get_coord_labels(reeds_build)
+
+        if rev_coord_labels is not None and reeds_coord_labels is not None:
+            reeds_build = reeds_build.sort_values(reeds_join_on)
+            reeds_sc_gids = reeds_build[reeds_join_on].values
+            reeds_coords = reeds_build[reeds_coord_labels]
+
+            rev_mask = (rev_sc[rev_join_on].isin(reeds_sc_gids))
+            rev_sc = rev_sc.sort_values(rev_join_on)
+            rev_coords = rev_sc.loc[rev_mask, rev_coord_labels]
+
+            diff = reeds_coords.values - rev_coords.values
+            diff = diff.flatten()
+
+            if diff.max() > threshold:

```

Comment:
I know I started this approach but wouldn't np.allclose be cleaner?

Reply:
implemented, worse error message though.